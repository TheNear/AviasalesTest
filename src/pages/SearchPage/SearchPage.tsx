import React from "react";
import style from "./SearchPage.module.css";
import Filter from "../../components/Filter/Filter";
import SortControls from "../../components/SortControls/SortControls";
import TicketsList from "../../components/TicketsList/TicketsList";
import { useTickets } from "../../hooks/useTickets";

const SearchPage: React.FC = () => {
  const { tickets, isLoading, errors, refreshTicketsList } = useTickets(5);

  return (
    <main className={style.container}>
      <Filter />
      <SortControls />
      <TicketsList
        isLoading={isLoading}
        hasError={errors}
        refreshTicketsList={refreshTicketsList}
        tickets={tickets.slice(0, 5)}
      />
    </main>
  );
};

export default SearchPage;
