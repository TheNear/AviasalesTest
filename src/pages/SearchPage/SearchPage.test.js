import React from "react";
import { shallow } from "enzyme";
import SearchPage from "./SearchPage";

describe("Test SearchPage component", () => {
  it("Render empty SearchPage", () => {
    const wrapper = shallow(<SearchPage />);
    expect(wrapper).toMatchSnapshot();
  });
});
