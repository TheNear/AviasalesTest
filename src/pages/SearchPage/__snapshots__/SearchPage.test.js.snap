// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test SearchPage component Render empty SearchPage 1`] = `
<main
  className="container"
>
  <Filter />
  <SortControls />
  <TicketsList
    hasError=""
    isLoading={false}
    refreshTicketsList={[Function]}
    tickets={Array []}
  />
</main>
`;
